### PR TITLE
[Frontend] - Product Info Component

### PR DESCRIPTION
**Pull request by TreshAnn**
## Ticket/s

- [Product Info Component](https://trello.com/c/3VtlllW7)
- [Product Gallery Component](https://trello.com/c/WkzgFy11)

## Changes

- Add ProductInfo Component
- Import ProductGallery Component
## Screenshot

image

ProductInfo

image

image

## How to test?

-  run **pnpm  run dev**
- After compilation is complete
- Insert to url http://localhost:5173/

Note: 
- You can edit the default values of the props as needed
- If any errors occur, try running `pnpm install` inside the **ui** directory.

### CODE REVIEW

**Review by SGVGen — DISMISSED** [review]
I think it's missing the Web Responsive:
Product - Info

**Comment by TreshAnn** [comment]
> I think it's missing the Web Responsive:

Done, sir @SGVGen. let me know if any further changes are needed. Thanks

**Review by sandrex-SGV — COMMENTED** [review]
Works with my end.

Just a suggestion, can you make the font size smaller as the device screen gets smaller?

Just use standard breakpoints for reference of device screen sizes. Thanks!

[Bootstrap Breakpoints](https://getbootstrap.com/docs/5.0/layout/breakpoints/)

**Review by markjoshuacarmen — APPROVED** [review]
looks nice!

File: packages/ui/product-info/productInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  StyledBox,
+  StyledContainer,
+} from '../product-info/style';
+
+interface IProductProps {
+  productName?: string;
+  ratingValue?: number;
+  productDescription?: string;
+  productPrice?: number;
+  merchantName?: string;
```
Comment (dandalandone) [inline packages/ui/product-info/productInfo.tsx]: why is this optional?

File: packages/ui/product-info/productInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+export function ProductInfo({
+  productName = 'Spicy Korean Fried Chicken',
+  ratingValue = 3,
+  productDescription = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.',
+  productPrice = 120,
+  merchantName = 'NaroShoppee',
```
Comment (dandalandone) [inline packages/ui/product-info/productInfo.tsx]: move default props to separate mock data file

File: packages/ui/product-info/style.ts [hunk 3]
```diff
@@ -0,0 +1,222 @@
+import { Box, Button, Container, Text } from '@mantine/core';
+import styled from 'styled-components';
+
+const maxMobileWidth = '976px';
+const sm = '600px';
+const md = '900px';
+const lg = '1200px';
```
Comment (dandalandone) [inline packages/ui/product-info/style.ts]: upon checking this sizes can be move to constant to be used to all styling

**Review by dandalandone — CHANGES_REQUESTED** [review]
Please check comments

File: packages/ui/product-info/productInfo.tsx (hunk 1 above)
Comment (TreshAnn) [reply packages/ui/product-info/productInfo.tsx]: Done, sir! thanks.

File: packages/ui/product-info/productInfo.tsx (hunk 2 above)
Comment (TreshAnn) [reply packages/ui/product-info/productInfo.tsx]: done, sir. please check, thanks.

File: packages/ui/product-info/style.ts (hunk 3 above)
Comment (TreshAnn) [reply packages/ui/product-info/style.ts]: done. please check. thanks!

File: packages/ui/Enum/enum.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export enum Breakpoints {
```
Comment (dandalandone) [inline packages/ui/Enum/enum.ts]: move to utils/screen-size.constant.ts

File: packages/ui/product-gallery/style.ts [hunk 5]
```diff
@@ -1,18 +1,26 @@
 import { Container } from '@mantine/core';
 import styled from 'styled-components';
 
-const maxMobileWidth = '976px';
-const sm = '600px';
+import { Breakpoints } from '../Enum/enum';
+
+const xs = Breakpoints.xs;
+const sm = Breakpoints.sm;
+const xl = Breakpoints.xl;
```
Comment (dandalandone) [inline packages/ui/product-gallery/style.ts]: no need for this, access it directly.

File: packages/ui/product-gallery/style.ts [hunk 6]
```diff
@@ -1,18 +1,26 @@
...
-  width: 772px;
-  height: 604px;
+  width: 700px;
+  height: 500px;
   margin: 0px;
+  padding: 0;
+  justify-content: center;
 
-  @media (max-width: ${maxMobileWidth}) {
+  @media (max-width: ${xl}) {
```
Comment (dandalandone) [inline packages/ui/product-gallery/style.ts]:
```suggestion
  @media (max-width: ${Breakpoints.xl}) {
```

File: packages/ui/product-gallery/style.ts [hunk 7]
```diff
@@ -1,18 +1,26 @@
...
+    width: auto;
+    height: 500px;
+  }
+  @media (max-width: ${sm}) {
```
Comment (dandalandone) [inline packages/ui/product-gallery/style.ts]:
```suggestion
  @media (max-width: ${Breakpoints.sm}) {
```

File: packages/ui/product-gallery/style.ts [hunk 8]
```diff
@@ -25,12 +33,7 @@ export const LeftColumn = styled.div`
     overflow-y: auto;
   }
 
-  @media (max-width: ${maxMobileWidth}) {
-    &:hover {
-      overflow-y: auto;
-    }
-  }
-  @media (max-width: ${sm}) {
+  @media (max-width: ${xl}) {
```
Comment (dandalandone) [inline packages/ui/product-gallery/style.ts]: same with others.

File: packages/ui/product-gallery/style.ts [hunk 9]
```diff
@@ -40,11 +43,10 @@ export const RightColumn = styled.div`
   display: flex;
   justify-content: center;
   align-items: center;
-  margin: 10px;
-  width: 604px;
+  width: 455px;
 
-  @media (max-width: ${sm}) {
-    display: none;
+  @media (max-width: ${xl}) {
```
Comment (dandalandone) [inline packages/ui/product-gallery/style.ts]: same with others

File: packages/ui/product-info/productInfo.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+  ratingValue: number;
+  productDescription: string;
+  productPrice: number;
+  merchantName: string;
+  productImg: string[];
+}
+
+interface Props {
+  data: IProductProps;
+}
```
Comment (dandalandone) [inline packages/ui/product-info/productInfo.tsx]: no need for this. use directly IProductProps

File: packages/ui/product-info/style.ts [hunk 11]
```diff
@@ -0,0 +1,262 @@
+import { Box, Button, Container, Text } from '@mantine/core';
+import styled from 'styled-components';
+
+import { Breakpoints } from '../Enum/enum';
+
+const xs = Breakpoints.xs;
+const sm = Breakpoints.sm;
+const md = Breakpoints.md;
+const lg = Breakpoints.lg;
+const xl = Breakpoints.xl;
```
Comment (dandalandone) [inline packages/ui/product-info/style.ts]: access it direclty

**Review by dandalandone — CHANGES_REQUESTED** [review]
See comments, thanks.

File: packages/ui/Enum/enum.ts (hunk 4 above)
Comment (TreshAnn) [reply packages/ui/Enum/enum.ts]: applied all the changes. Please check, Thanks.

**Review by dandalandone — APPROVED** [review]
LGTM!

**Pull request merged**